test/e2e/external_gateways.go

```diff
@@ -802,7 +802,7 @@ var _ = ginkgo.Describe("External Gateway", func() {
 				ginkgo.Entry("IPV6 udp", &addressesv6, "udp"),
 				ginkgo.Entry("IPV6 tcp", &addressesv6, "tcp"))
 
-			ginkgo.DescribeTable("ExternalGWPod annotation: Should validate conntrack entry deletion for TCP/UDP traffic via multiple external gateways a.k.a ECMP routes", func(addresses *gatewayTestIPs, protocol string) {
+			ginkgo.DescribeTable("ExternalGWPod annotation: Should validate conntrack entry deletion for TCP/UDP traffic via multiple external gateways a.k.a ECMP routes", func(addresses *gatewayTestIPs, protocol string, deletePod bool) {
```

Comment:
Wouldn't it be better to split this into its own table/use case? one for annotation deletion and another one for pod deletion. The flow is mostly the same, but the breakdown can help to write a more specific description of what the test does in case the test fails (it failed while deleting annotation or it failed while deleting the pod). WDYT?

Reply:
You can distinguish the tests even now:
1. test names will be different 
2. ginkgo.By will log different steps
I also find it much easier to understand the difference between test sequences when it shows which step exactly differs, e.g. vs having a full copy of this test with 3 different lines

Reply:
Honestly it's a personal preference between readability vs reusability. I tend to prefer readability in tests even if that means replicating code that can't be extracted as functions (perhaps in this case it could be, but that's another story).

On the other hand, `true` or `false` is not clear what they mean as arguments without understanding the logic inside the test. I'd recommend to use constants whose names define their behavior rather than booleans as arguments for what they mean, e.g. `BY_DELETE_ANNOTATION`, `BY_DELETE_GW_POD`.

It's a cosmetic comment since this logic will go away, eventually I hope 😄 .